instrumentation/pulsar/pulsar-2.8/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/pulsar/v2_8/telemetry/PulsarSingletons.java

```diff
@@ -64,47 +64,61 @@ private static Instrumenter<PulsarRequest, Void> createConsumerReceiveInstrument
     MessagingAttributesGetter<PulsarRequest, Void> getter =
         PulsarMessagingAttributesGetter.INSTANCE;
 
-    return Instrumenter.<PulsarRequest, Void>builder(
-            TELEMETRY,
-            INSTRUMENTATION_NAME,
-            MessagingSpanNameExtractor.create(getter, MessageOperation.RECEIVE))
-        .addAttributesExtractor(
-            createMessagingAttributesExtractor(getter, MessageOperation.RECEIVE))
-        .addAttributesExtractor(
-            NetClientAttributesExtractor.create(new PulsarNetClientAttributesGetter()))
-        .buildConsumerInstrumenter(MessageTextMapGetter.INSTANCE);
+    boolean enabled = ExperimentalConfig.get().messagingReceiveInstrumentationEnabled();
+    InstrumenterBuilder<PulsarRequest, Void> builder =
+        Instrumenter.<PulsarRequest, Void>builder(
+                TELEMETRY,
+                INSTRUMENTATION_NAME,
+                MessagingSpanNameExtractor.create(getter, MessageOperation.RECEIVE))
+            .addAttributesExtractor(
+                createMessagingAttributesExtractor(getter, MessageOperation.RECEIVE))
+            .addAttributesExtractor(
+                NetClientAttributesExtractor.create(new PulsarNetClientAttributesGetter()))
+            .setEnabled(enabled);
+    if (enabled) {
+      builder.addSpanLinksExtractor(new PulsarRequestSpanLinksExtractor(PROPAGATOR));
+    }
+    return builder.buildInstrumenter(SpanKindExtractor.alwaysConsumer());
```

Comment:
Nit: you don't need to have an extra check for enabled here, if it's not enabled the instrumenter won't generate any telemetry anyway, so you can simply always add the links extractor:
```suggestion
            .setEnabled(enabled);
            .addSpanLinksExtractor(new PulsarRequestSpanLinksExtractor(PROPAGATOR));
            .buildInstrumenter(SpanKindExtractor.alwaysConsumer());
```